src/controllers/PasswordController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+class PasswordController extends ApiController
+{
+    protected $oauthModel;
+
+    /**
+     * Handles a request.
+     *
+     * @param Request $request the requeste
+     * @param         $db      the database.
+     *
+     * @return mixed
+     */
+    public function handle(Request $request, $db)
+    {   
+        $this->oauthModel = $request->getOauthModel($db);
+
+        // only POST is implemented so far
+        if ($request->getVerb() == 'POST') {
+            return $this->postAction($request, $db);
+        }
+
+        return false;
+    }
+
+    /**
+     * Handles a POST request.
+     *
+     *
+     * @param Request $request the request.
+     * @param         $db      the database.
+     *
+     * @return mixed
+     */
+    public function postAction(Request $request, $db)
+    {
+        if (!isset($request->user_id)) {
+            throw new Exception("You must be logged in to change your password", 400);
+        }
+
+        $userId    = $request->getUserId();
```

Comment:
We should probably have a policy on whether we're using `$request->user_id` or `$request->getUserId()`. We appear to use the property directly everywhere else. Discussion required.